core/domain/opportunity_services.py

```diff
@@ -0,0 +1,380 @@
+    the chapters of a story of a topic.
+
+    Args:
+        exp_id: str. The id of the exploration which is needed to be checked.
+
+    Returns:
+        bool. Whether the given exp_id belongs to the curated explorations.
+    """
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(
+        exp_id, strict=False)
+
+    return True if model is not None else False
+
+
+def get_exploration_opportunity_summary_from_model(model):
+    """Returns the ExplorationOpportunitySummary object out of the model.
+
+    Args:
+        model: ExplorationOpportunitySummaryModel. The exploration opportunity
+            summary model.
+
+    Returns:
+        ExplorationOpportunitySummary. The corresponding
+        ExplorationOpportunitySummary object.
+    """
+
+    return opportunity_domain.ExplorationOpportunitySummary(
+        model.id, model.topic_id, model.topic_name, model.story_id,
+        model.story_title, model.chapter_title, model.content_count,
+        model.incomplete_translation_languages, model.translation_counts,
+        model.need_voice_artist_in_languages,
+        model.assigned_voice_artist_in_languages)
+
+
+def _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list):
+    """Stores multiple ExplorationOpportunitySummary into datastore as a
+    ExplorationOpportunitySummaryModel.
+
+    Args:
+        exploration_opportunity_summary_list: list(
+            ExplorationOpportunitySummary). A list of exploration opportunity
+            summary object.
+    """
+    exploration_opportunity_summary_model_list = []
+    for opportunity_summary in exploration_opportunity_summary_list:
+        model = opportunity_models.ExplorationOpportunitySummaryModel(
+            id=opportunity_summary.id,
+            topic_id=opportunity_summary.topic_id,
+            topic_name=opportunity_summary.topic_name,
+            story_id=opportunity_summary.story_id,
+            story_title=opportunity_summary.story_title,
+            chapter_title=opportunity_summary.chapter_title,
+            content_count=opportunity_summary.content_count,
+            incomplete_translation_languages=(
+                opportunity_summary.incomplete_translation_languages),
+            translation_counts=opportunity_summary.translation_counts,
+            need_voice_artist_in_languages=(
+                opportunity_summary.need_voice_artist_in_languages),
+            assigned_voice_artist_in_languages=(
+                opportunity_summary.assigned_voice_artist_in_languages)
+        )
+
+        exploration_opportunity_summary_model_list.append(model)
+
+    opportunity_models.ExplorationOpportunitySummaryModel.put_multi(
+        exploration_opportunity_summary_model_list)
+
+
+def add_new_exploration_opportunities(story, exp_ids):
+    """Adds new exploration opportunity into the model.
+
+    Args:
+        story: Story. A story object.
```

Comment:
Description is not sufficient. Perhaps: The story which contains the exploration IDs.

Reply:
Also, better to pass a story_id rather than a Story object. Then, in this function, retrieve the story using the story ID.

Reply:
Probably best to merge in the story + topic fetcher PR before this one. Can you coordinate with @aks681 to get that in ASAP?